### PR TITLE
Remove useless RSOperator & refactor function parseTiCompareExpr

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233 

Problem Summary:

### What is changed and how it works?

In TiDB, `NotIn -> Not(In)`, `NotLike -> Not(Like)`, so remove `NotIn` and `NotLike`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp [hunk 1]
```diff
@@ -106,47 +94,53 @@ ColumnDefine getColumnDefineForColumnExpr(const tipb::Expr & expr, const ColumnD
...
+                "{} with {} children is not supported",
+                tipb::ScalarFuncSig_Name(expr.sig()),
+                expr.children_size()),
             false);
 
-    /// Only support `column` `op` `literal`, and `column` in (literal1, literal2, ...) now.
+    // Support three types of expression:
+    // 1. `column` `op` `literal`
+    // 2. `literal` `op` `column`
+    // 3. `column` in (literal1, literal2, ...)
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp:116]:
```suggestion
    // 3. `column` in (literal1, literal2, ...); Note that the children size of `col in (1,2)` is 3
```

File: dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp [hunk 2]
```diff
@@ -106,47 +94,53 @@ ColumnDefine getColumnDefineForColumnExpr(const tipb::Expr & expr, const ColumnD
...
     for (const auto & child : expr.children())
     {
         if (isColumnExpr(child))
+        {
             is_timestamp_column = (child.field_type().tp() == TiDB::TypeTimestamp);
+            break;
+        }
     }
-    for (int32_t child_idx = 0; child_idx < expr.children_size(); ++child_idx)
+    int column_child_idx = -1;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp:130]:
```suggestion
    int column_expr_child_idx = -1;
```

File: dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp [hunk 3]
```diff
@@ -106,47 +94,53 @@ ColumnDefine getColumnDefineForColumnExpr(const tipb::Expr & expr, const ColumnD
...
         if (isColumnExpr(child))
         {
+            // If iter to the second column, return unsupported.
+            if (child_idx == 0 || column_child_idx == -1)
+                column_child_idx = child_idx;
+            else
+                return createUnsupported(
+                    expr.ShortDebugString(),
+                    "Multiple ColumnRef in expression is not supported",
+                    false);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp:143]: Why we need `child_idx == 0 || column_child_idex == -1` here? Is it OK to only check `column_child_index == -1`?

File: dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp (hunk 1 above)
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp]: https://github.com/pingcap/tiflash/pull/8212/commits/0bcc39e0ef48ff9d66c804cc5ba6661b4a22f062

change to 
```
    // 1. op(column, literal), in sql: column op literal
    // 2. op(literal, column), in sql: literal op column
    // 3. in(coloumn, literal1, literal2, ...), in sql: column in (literal1, literal2, ...)
    // op is one of: =, !=, >, >=, <, <=
```

File: dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp (hunk 3 above)
Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp]: Fixed in https://github.com/pingcap/tiflash/pull/8212/commits/b38399f882107aa97032c04e04713a4166aa0bd3

File: dbms/src/Storages/DeltaMerge/tests/gtest_dm_minmax_index.cpp [hunk 4]
```diff
@@ -2114,65 +2114,124 @@ try
...
-    {
-        // make a not in filter, check not in (NULL, 1)
-        auto filter = createNotIn(attr("Nullable(Int64)"), {Field(), Field(static_cast<Int64>(1))});
-        ASSERT_EQ(filter->roughCheck(0, 1, param)[0], RSResult::Some);
-    }
-    {
-        // make a not in filter, check not in (3)
-        auto filter = createNotIn(attr("Nullable(Int64)"), {Field(static_cast<Int64>(3))});
-        ASSERT_EQ(filter->roughCheck(0, 1, param)[0], RSResult::All);
-    }
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/tests/gtest_dm_minmax_index.cpp:2131]: turn these into cases of `createNote(createIn(....))` ?

Comment (Lloyd-Pottiger) [reply dbms/src/Storages/DeltaMerge/tests/gtest_dm_minmax_index.cpp]: Good idea! Fixed in https://github.com/pingcap/tiflash/pull/8212/commits/2fa3304b099fe1607d648814eb18774724febb33

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**